exo/inference/inference_engine.py

```diff
@@ -29,6 +29,10 @@ async def infer_prompt(self, request_id: str, shard: Shard, prompt: str) -> np.n
     output_data = await self.infer_tensor(request_id, shard, tokens)
     return output_data 
 
+  @abstractmethod
+  async def preload_model(self, shard: Shard) -> None:
+    pass
```

Comment:
Need a `preload_model` implementation for `TinygradDynamicShardInferenceEngine` and `DummyInferenceEngine`. You can test with `exo --inference-engine tinygrad`

Reply:
done